internal/middleware/logging.go

```diff
@@ -2,65 +2,103 @@ package middleware
 
 import (
 	"context"
-	"time"
+	"encoding/json"
 
 	"github.com/openfga/openfga/pkg/logger"
 	serverErrors "github.com/openfga/openfga/pkg/server/errors"
+	openfgapb "go.buf.build/openfga/go/openfga/api/openfga/v1"
+	"go.opentelemetry.io/otel/trace"
 	"go.uber.org/zap"
 	"google.golang.org/grpc"
 )
 
+const (
+	grpcServiceKey  = "grpc_service"
+	grpcMethodKey   = "grpc_method"
+	requestIDKey    = "request_id"
+	traceIDKey      = "trace_id"
+	rawRequestKey   = "raw_request"
+	rawResponseKey  = "raw_response"
+	publicErrorKey  = "public_error"
+	grpcErrorKey    = "grpc_error"
```

Comment:
I also recommend we introduce this as `grpc_code` and it's values will be things like `0`, `Code(2022)` (for our custom error codes 😢 ), `InvalidArgument`, etc... 


Can we also introduce `grpc_type` to capture the kind of grpc request.

```
grpc_type: {
    enum: ["unary", "client_stream", "server_stream", "bidi_stream"]
}
```

Reply:
grpc_code and grpc_type introduced. Thanks!